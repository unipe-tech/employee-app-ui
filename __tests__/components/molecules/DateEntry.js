import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { describe, expect, test } from "@jest/globals";
import DateEntry from "../../../components/molecules/DateEntry";

describe("DateEntry", () => {
  test("renders UI correctly", () => {
    const mockSetVal = jest.fn();
    const tree = render(<DateEntry setval={mockSetVal} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
