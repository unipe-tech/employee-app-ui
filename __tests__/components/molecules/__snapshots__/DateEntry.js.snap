// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateEntry renders UI correctly 1`] = `
Array [
  <Text
    style={
      Object {
        "color": "#808080",
        "fontFamily": "Montserrat-Regular",
        "fontSize": 14,
        "lineHeight": 18,
        "marginTop": 59.21052631578947,
      }
    }
  />,
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      keyboardType="numeric"
      maxLength={2}
      onChangeText={[Function]}
      placeholder="DD"
      style={
        Object {
          "borderBottomWidth": 1,
          "fontFamily": "Montserrat-Regular",
          "fontSize": 14,
          "height": 78.94736842105263,
          "lineHeight": 18,
          "textAlign": "center",
          "width": 78.94736842105263,
        }
      }
      value=""
    />
    <View
      style={
        Object {
          "alignSelf": "center",
          "width": 59.21052631578947,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontFamily": "Montserrat-Bold",
            "fontSize": 30,
            "lineHeight": 36,
          }
        }
      >
        -
      </Text>
    </View>
    <TextInput
      keyboardType="numeric"
      maxLength={2}
      onChangeText={[Function]}
      placeholder="MM"
      style={
        Object {
          "borderBottomWidth": 1,
          "fontFamily": "Montserrat-Regular",
          "fontSize": 14,
          "height": 78.94736842105263,
          "lineHeight": 18,
          "textAlign": "center",
          "width": 78.94736842105263,
        }
      }
      value=""
    />
    <View
      style={
        Object {
          "alignSelf": "center",
          "width": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 30,
          }
        }
      >
        -
      </Text>
    </View>
    <TextInput
      keyboardType="numeric"
      maxLength={4}
      onChangeText={[Function]}
      placeholder="YYYY"
      style={
        Object {
          "borderBottomWidth": 1,
          "fontFamily": "Montserrat-Regular",
          "fontSize": 14,
          "height": 78.94736842105263,
          "lineHeight": 18,
          "textAlign": "center",
          "width": 78.94736842105263,
        }
      }
      value=""
    />
  </View>,
]
`;
